packages/manager/cypress/e2e/core/cloudpulse/cloudpulse-validation.spec.ts

```diff
@@ -0,0 +1,468 @@
+      mockGetCloudPulseDashboard(id, dashboard); // Mock fetching a specific dashboard by ID
+      mockCreateCloudPulseJWEToken(serviceType); // Mock creating a JWE token for authentication
+      mockGetRegions([mockRegion]); // Mock fetching the regions available
+
+      // Step 2: Intercept the API call for fetching DB clusters and simulate a 400 Bad Request error response.
+      cy.intercept('GET', apiMatcher(`databases/instances*`), {
+        statusCode: statusCode, // Use the status code defined in the loop
+        body: {
+          errors: [
+            {
+              reason: 'Bad Request', // Specify the error reason
+            },
+          ],
+        },
+      }).as('fetchCluster'); // Alias for the intercept to be used later
+
+      // Step 3: Navigate to the CloudPulse monitoring page after logging in.
+      cy.visitWithLogin('monitor/cloudpulse');
+
+      // Step 4: Wait for the services and dashboard API calls to resolve before proceeding.
+      cy.wait(['@fetchServices', '@fetchDashboard']);
+
+      // Step 5: Select a dashboard from the autocomplete input. Verify that the input is visible before typing.
+      ui.autocomplete
+        .findByLabel('Dashboard')
+        .should('be.visible') // Check that the dashboard input is visible
+        .type(`${dashboardName}{enter}`) // Type the dashboard name and press enter
+        .should('be.visible'); // Ensure the dashboard input remains visible
+
+      // Step 6: Select a Node Type from the autocomplete input. Verify visibility before typing.
+      ui.autocomplete
+        .findByLabel('Node Type')
+        .should('be.visible') // Check that the Node Type input is visible
+        .type(`${nodeType}{enter}`); // Type the node type and press enter
+
+      // Step 7: Select a region from the dropdown. Click and type the region name.
+      ui.regionSelect.find().click().type(`${region}{enter}`); // Click on the region dropdown and select the region
+
+      // Step 8: Select a Database Engine from the autocomplete input. Verify visibility before typing.
+      ui.autocomplete
+        .findByLabel('Database Engine')
+        .should('be.visible') // Check that the Database Engine input is visible
+        .type(`${engine}{enter}`) // Type the engine name and press enter
+        .should('be.visible'); // Ensure the engine input remains visible
+
+      // Step 9: Assert that the error message for fetching Database Clusters is displayed correctly.
+      cy.get('[data-qa-textfield-error-text="Database Clusters"]') // Select the error message element
+        .should('be.visible') // Check that the error message is visible
+        .invoke('text') // Get the text content of the error message
+        .then((text) => {
+          // Step 10: Verify that the displayed error message matches the expected message.
+          expect(text.trim()).to.equal(`Failed to fetch Database Clusters`); // Check if the message is as expected
```

Comment:
why trim? same at all places

Reply:
To handle extra spaces in the text retrieved from the UI

Reply:
I have Addresed  this issue.